docs/flow.rst

```diff
@@ -0,0 +1,235 @@
+===================
+Resolwe Flow Design
+===================
+
+The Resolwe Flow dataflow engine comprises the execution framework and the
+management layers which make up the internal data model and facilitate
+dependency resolution, permissions enforcement and process filtering.
+
+Overview
+========
+
+Execution Framework
+-------------------
+
+Flow consists of a number of active services which need to be running before job
+execution can proceed.
+
+The core message transport and coordination facility, as currently used, is
+`Redis`_. It serves as the central status hub for keeping track of shared
+dynamic information about the manager instances, and as a contact point for
+other parts of the framework, such as the executors. These connect to well-known
+'channels' (specially named Redis list objects), into which they can deposit
+JSON-formatted messages and commands.
+
+.. _Redis: https://redis.io
+
+Flow's execution engine, the 'manager', is an independent service which runs as
```

Comment:
Resolwe generally is running low on naming vocabulary because there are so many things in it that handle execution in some way yet are completely different. This summary is wrong. I've changed execution engine to execution manager.